chain/epoch_manager/src/lib.rs

```diff
@@ -306,8 +306,13 @@ impl EpochManager {
                 )?;
             } else {
                 let prev_block_info = self.get_block_info(&block_info.prev_hash)?.clone();
+                let epoch_info = self.get_epoch_info(&prev_block_info.epoch_id)?;
+                // Keep slashed from previous block if they are still in the epoch info stake change
```

Comment:
How can it be that someone is slashed but they "do not return the stake"? I thought stake is taken away from them automatically.

Reply:
It is and that's exactly what caused the invariant to fail. This part fixes that.

Reply:
I still don't understand what it means "still slashed for not returning stake". How can someone not return the stake?

Reply:
It's related to how we return stake. If you are slashed it's a special case because your stake is taken directly, which means it should not go through the normal procedure of how stake is returned. That was the cause the bug.

Reply:
> If you are slashed it's a special case because your stake is taken directly, which means it should not go through the normal procedure of how the stake is returned.

I understand, but I would expect the slashed stake to be taken away automatically from the slashed user. The comment makes it sound like the slashes user needs to return their stake themselves. Is it the case?

Reply:
No that's not the case. Maybe the comment can be improved. I think Illia meant something like "the system do not return their stake"